### PR TITLE
Update TS docker build information

### PR DESCRIPTION
**Pull request by bergundy**

### CODE REVIEW

**Review by mjameswh — APPROVED** [review]
@bergundy While we're at this, maybe also update version on all Docker image references to `:20`.

**Comment by bergundy** [comment]
@mjameswh PTAL, I also added advice to refer to the production sample for bundling the workflow code and saving it to a file.

File: docs-src/typescript/how-to-run-a-worker-on-docker.md [hunk 1]
```diff
@@ -31,7 +37,7 @@ RUN npm install --only=production \
 CMD ["build/worker.js"]
 ```
 
-For smaller images and/or more secure deployments, it is also possible to use `-slim` Docker image variants (like `node:18-bullseye-slim`) or `distroless/nodejs` Docker images (like `gcr.io/distroless/nodejs:18`) with the following caveats.
+For smaller images and/or more secure deployments, it is also possible to use `-slim` Docker image variants (like `node:20-bullseye-slim`) or `distroless/nodejs` Docker images (like `gcr.io/distroless/nodejs:20`) with the following caveats.
```
Comment (mjameswh) [inline docs-src/typescript/how-to-run-a-worker-on-docker.md:40]: Looks like the distroless project changed their URL structures starting with Node 20.

```suggestion
For smaller images and/or more secure deployments, it is also possible to use `-slim` Docker image variants (like `node:20-bullseye-slim`) or `distroless/nodejs` Docker images (like `gcr.io/distroless/nodejs20-debian11`) with the following caveats.
```

File: docs-src/typescript/how-to-run-a-worker-on-docker.md [hunk 2]
```diff
@@ -82,7 +88,7 @@ RUN npm install --only=production \
 
 # -- RESULTING IMAGE --
 
-FROM gcr.io/distroless/nodejs:18
+FROM gcr.io/distroless/nodejs:20
```
Comment (mjameswh) [inline docs-src/typescript/how-to-run-a-worker-on-docker.md:91]:
```suggestion
FROM gcr.io/distroless/nodejs20-debian11
```

**Review by mjameswh — APPROVED** [review]
Good, just need to fix URL for distroless node 20

**Comment by bergundy** [comment]
@mjameswh I made a few changes since your review, PTAL.

**Review by flossypurse — APPROVED** [review]

File: docs-src/typescript/how-to-run-a-worker-on-docker.md [hunk 3]
```diff
@@ -9,29 +9,38 @@ tags:
   - typescript
 ---
 
+:::note
+
+Before proceeding, we recommend referring to our [production sample](https://github.com/temporalio/samples-typescript/tree/main/production),
```
Comment (mjameswh) [inline docs-src/typescript/how-to-run-a-worker-on-docker.md:14]: I'd reverse that sentence. Our recommendation is really that they prepare the workflow bundle ahead-of-time, not that they look at the production sample.

```suggestion
To improve worker startup time, we recommend that workflow bundles be prepared ahead-of-time. See our [production sample](https://github.com/temporalio/samples-typescript/tree/main/production) for details.
```

File: docs-src/typescript/how-to-run-a-worker-on-docker.md [hunk 4]
```diff
@@ -9,29 +9,38 @@ tags:
...
 ```dockerfile
-FROM node:18-bullseye
+FROM node:20-bullseye
 
+# For better cache utilization, copy package.json and lock file first and install the dependencies before copying the
+# rest of the application and building.
 COPY . /app
 WORKDIR /app
 
+# Alternatively, run npm ci, which installs only dependencies specified in the lock file and is generally faster.
```
Comment (mjameswh) [inline docs-src/typescript/how-to-run-a-worker-on-docker.md:36]: I think at this point, it should be reasonable to recommend this instead:

```
npm ci --omit dev
```
`--omit` was introduced in NPM 8, which itself got introduced in 16.11.0.

**Review by mjameswh — APPROVED** [review]
A few more suggestions, but still good.

File: docs-src/typescript/how-to-run-a-worker-on-docker.md (hunk 4 above)
Comment (bergundy) [reply docs-src/typescript/how-to-run-a-worker-on-docker.md:36]: `--omit dev` would require a separate build step, I'd avoid that for now.

File: docs-src/typescript/how-to-run-a-worker-on-docker.md (hunk 3 above)
Comment (bergundy) [reply docs-src/typescript/how-to-run-a-worker-on-docker.md]: Changed to more or less what you're suggesting.

**Pull request merged**